apps/dashboard/src/components/auth/questionnaire-form.tsx

```diff
@@ -0,0 +1,199 @@
+import { Button } from '@/components/primitives/button';
+import { CardDescription, CardTitle } from '@/components/primitives/card';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/primitives/select';
+import React, { useState } from 'react';
+import { StepIndicator } from './shared';
+import { JobTitleEnum, jobTitleToLabelMapper, OrganizationTypeEnum, CompanySizeEnum } from '@novu/shared';
+import { useForm, Controller } from 'react-hook-form';
+import { updateClerkOrgMetadata } from '../../api/organization';
+import { hubspotCookie } from '../../utils/cookies';
+import { identifyUser } from '../../api/telemetry';
+import { useTelemetry } from '../../hooks';
+import { TelemetryEvent } from '../../utils/telemetry';
+import { useNavigate } from 'react-router-dom';
+import { ROUTES } from '../../utils/routes';
+
+interface QuestionnaireFormData {
+  jobTitle: JobTitleEnum;
+  organizationType: OrganizationTypeEnum;
+  companySize?: CompanySizeEnum;
+}
+
+export function QuestionnaireForm() {
+  const { control, watch, handleSubmit } = useForm<QuestionnaireFormData>();
+  const track = useTelemetry();
+  const navigate = useNavigate();
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const selectedJobTitle = watch('jobTitle');
+  const selectedOrgType = watch('organizationType');
+  const companySize = watch('companySize');
+
+  const shouldShowCompanySize =
+    (selectedOrgType === OrganizationTypeEnum.COMPANY || selectedOrgType === OrganizationTypeEnum.AGENCY) &&
+    !!selectedJobTitle;
+
+  const isFormValid = React.useMemo(() => {
+    if (!selectedJobTitle || !selectedOrgType) return false;
+    if (shouldShowCompanySize && !companySize) return false;
+
+    return true;
+  }, [selectedJobTitle, selectedOrgType, shouldShowCompanySize, companySize]);
+
+  const onSubmit = async (data: QuestionnaireFormData) => {
+    setIsSubmitting(true);
+
+    try {
+      await updateClerkOrgMetadata({
+        companySize: data.companySize,
+        jobTitle: data.jobTitle,
+        organizationType: data.organizationType,
+      });
+
+      const hubspotContext = hubspotCookie.get();
+
+      await identifyUser({
+        jobTitle: data.jobTitle,
+        pageUri: window.location.href,
+        pageName: 'Create Organization Form',
+        hubspotContext: hubspotContext || '',
+        companySize: data.companySize,
+        organizationType: data.organizationType,
+      });
```

Comment:
Please move these under the `useMutation` and handle the error case with `onError`. You won't also need to track the pending state because you could use `const { isPending } = useMutation(...` flag.

Reply:
Awesome! Missed that we use react query for mutations. Thanks